### DISCUSSION TITLE
Help for codemagic

### DISCUSSION THREAD

**Discussion by JulianHSC** [codemagic-ci-cd/codemagic-docs#2429] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

React Native

### Steps to reproduce

I got the following steps but I can't find the location to download the yarn file. Can someone advice what is happening?

Log in in to Codemagic.
In the Applications dashboard, search for your project and go to its Settings.
Now, go to the tab Advanced configuration .
From here you can download the codemagic. yaml file for this project by clicking on the Download configuration button.
![React native basic world yarn problem](https://github.com/codemagic-ci-cd/codemagic-docs/assets/31162511/9f8372dd-7721-416a-9d6e-2d8fac24660b)

### Expected results

I should have ios .iap file created / generated. I don't know what is the build ID. No information anywhere.

### Actual results

This is what I get instead:
Get started by following one of our React Native quick start guides
Follow the [React Native](https://docs.codemagic.io/yaml-quick-start/building-a-react-native-app/?utm_source=app&utm_campaign=yaml-quickstart) guide to configure your project and create the codemagic.yaml file.
Save the codemagic.yaml file to the project root folder.
Push the added file to the repository. From command line:
git add codemagic.yaml
git commit -m 'Add first workflow'
git push
Click the Check for configuration file button in the top right corner.

**Comment by himesh-cm** [comment]
Hi @JulianHSC,
You can find build id under your build details page's 'ID' section.  ﻿Navigate to the 'Builds' section on the left-hand menu, select the build you want to view, and click the copy icon.

**Comment by JulianHSC** [comment]
This is what I see when i follow your steps. I don't see any IDs. More help required.

![basic world codemagic build](https://github.com/codemagic-ci-cd/codemagic-docs/assets/31162511/f49205fa-3a28-4bf0-9daa-d30da93ad9aa)

**Reply by himesh-cm** [reply]
Hi @JulianHSC, you should see it on your build details page instead of the builds page.
here is the url format for your reference
https://codemagic.io/app/xxxxxxx/build/xxxxxxx

**Comment by JulianHSC** [comment]
i tried this https://codemagic.io/app/basicworld/build/basicworld and the page is not found. Is this what you mean?

As an update I tried this and it didn't work:
C:\Z-Android\RN_Text>npx expo prebuild
Need to install the following packages:
expo@50.0.14
Ok to proceed? (y) Y
npm WARN deprecated @babel/plugin-proposal-class-properties@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-class-properties instead.
npm WARN deprecated @babel/plugin-proposal-optional-catch-binding@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-optional-catch-binding instead.
npm WARN deprecated @babel/plugin-proposal-numeric-separator@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-numeric-separator instead.
npm WARN deprecated @babel/plugin-proposal-nullish-coalescing-operator@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-nullish-coalescing-operator instead.
npm WARN deprecated @npmcli/move-file@1.1.2: This functionality has been moved to @npmcli/fs
npm WARN deprecated @babel/plugin-proposal-optional-chaining@7.21.0: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-optional-chaining instead.
npm WARN deprecated @babel/plugin-proposal-object-rest-spread@7.20.7: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-object-rest-spread instead.
npm WARN deprecated @babel/plugin-proposal-logical-assignment-operators@7.20.7: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-logical-assignment-operators instead.
npm WARN deprecated @babel/plugin-proposal-async-generator-functions@7.20.7: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-async-generator-functions instead.

CommandError: Cannot determine which native SDK version your project uses because the module expo is not installed. Please install it with yarn add expo and try again.

C:\Z-Android\RN_Text>git add codemagic.yaml
fatal: pathspec 'codemagic.yaml' did not match any files

**Reply by himesh-cm** [reply]
Hi @JulianHSC , The issue seems to be with node-modules packages not being installed.

Please open a new thread along with your build-id
You can find it under your build details page's 'ID' section.  ﻿Navigate to the 'Builds' section on the left-hand menu, select the build you want to view, and click the copy icon.

**Comment by himesh-cm** [comment]
I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.

**Reply by JulianHSC** [reply]
I've been busy with other issues and hadn't had the time to check. Please check my screenshot again. There's a build page but there's no ID section. What can I do now? I've also checked all the drop down menus in the page but no IDs of any kind is given.
![codemagic build page](https://github.com/codemagic-ci-cd/codemagic-docs/assets/31162511/9f13101c-d939-4a08-a809-d545e6c7ac4e)
